visualization/app/codeCharta/util/algorithm/streetLayout/streetViewHelper.ts

```diff
@@ -0,0 +1,93 @@
+import { Vector3 } from "three"
+import { CodeMapNode, Node, State } from "../../../codeCharta.model"
+import { getMarkingColor, isLeaf } from "../../codeMapHelper"
+import { getBuildingColor, getIncomingEdgePoint, isNodeFlat, isVisible, TreeMapHelper } from "../treeMapLayout/treeMapHelper"
+
+function calculateSize(node: CodeMapNode, metricName: string) { // TODO if it is same as countNodes in treeMapHelper.ts
+	let totalSize = node.attributes[metricName] || 0
+
+	if (totalSize === 0 && node.children && node.children.length > 0) {
+		for (const child of node.children) {
+			totalSize += calculateSize(child, metricName)
+		}
+	}
+	return totalSize
+}
+
+function mergeDirectories(node: CodeMapNode, metricName: string): CodeMapNode {
+	let mergedNode = node
+	for (const child of node.children) {
+		if (!isLeaf(child)) {
+			const nodeSize = calculateSize(node, metricName)
+			const childSize = calculateSize(child, metricName)
```

Comment:
Please only calculate the `nodeSize` once.

```suggestion
	let mergedNode = node
	const nodeSize = calculateSize(node, metricName)
	if (nodeSize === 0) {
		return mergedNode
	}
	for (const child of node.children) {
		if (!isLeaf(child)) {
			const childSize = calculateSize(child, metricName)
			if (childSize === 0)
				continue
```

Reply:
Sure thing :)